### PR TITLE
[iOS/Catalyst] Correctly render ContentPage brushes background

### LINKED ISSUE #11714: Impossible to add the background gradients feature on ContentPage iOS
### Steps to Reproduce

Add this code to any of the content page

` <ContentPage.Background>
        <LinearGradientBrush EndPoint="0,1">
            <GradientStop Color="Yellow"
                          Offset="0.1" />
            <GradientStop Color="Green"
                          Offset="1.0" />
        </LinearGradientBrush>
    </ContentPage.Background>
    <ContentPage.Content>
        <Label Text="I love MAUI"
               VerticalOptions="CenterAndExpand"
               HorizontalOptions="CenterAndExpand"/>
    </ContentPage.Content>`

### Link to public reproduction project repository

https://github.com/samirgcofficial/GradientBugMAUI

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 14

### Did you find any workaround?

https://gist.github.com/vhugogarcia/b49af71df1e339626f758479d836b45a

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Correctly render ContentPage brushes background on iOS/Catalyst.

Captura de pantalla 2022-11-29 a las 10 55 29
Captura de pantalla 2022-11-29 a las 10 45 23

### Issues Fixed

Fixes #11714

### CODE REVIEW

File: src/Core/src/Platform/iOS/ViewExtensions.cs [hunk 1]
```diff
@@ -264,14 +265,24 @@ public static void UpdateBackgroundLayerFrame(this UIView view)
 
 			var layer = view.Layer;
 
+			UpdateBackgroundLayerFrame(layer, view.Bounds);
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/iOS/ViewExtensions.cs:268]: Not in 100% of the cases we will have the background layer as a sublayer of the main one. The main change is here, use recursion to get the background layer.

**Comment by jsuarezruiz** [comment]
This changes also fixes #10445 on iOS/Catalyst.
Captura de pantalla 2022-11-29 a las 11 14 31

File: src/Core/src/Platform/iOS/ViewExtensions.cs [hunk 2]
```diff
@@ -264,14 +265,24 @@ public static void UpdateBackgroundLayerFrame(this UIView view)
...
+
+		static void UpdateBackgroundLayerFrame(CALayer layer, CGRect bounds)
+		{
 			if (layer == null || layer.Sublayers == null || layer.Sublayers.Length == 0)
 				return;
 
 			foreach (var sublayer in layer.Sublayers)
 			{
-				if (sublayer.Name == BackgroundLayerName && sublayer.Frame != view.Bounds)
+				UpdateBackgroundLayerFrame(sublayer, bounds);
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/ViewExtensions.cs:278]: Does this change make the change on this PR no longer necessary?

https://github.com/dotnet/maui/pull/7925/files#diff-cf56310b45ff03f77e3aa0ee6c5d9b717ea28d750ce396f1b50148ccbcbed4f8R138

Comment (jsuarezruiz) [reply src/Core/src/Platform/iOS/ViewExtensions.cs:278]: Still is necessary. We are changing the method to search recursively but we still need to invoke it with Frame changes (resize, change orientation etc).

File: src/Core/src/Platform/iOS/ViewExtensions.cs [hunk 3]
```diff
@@ -264,14 +265,24 @@ public static void UpdateBackgroundLayerFrame(this UIView view)
...
+			}
+
+			foreach (var sublayer in layer.Sublayers)
```
Comment (hartez) [inline src/Core/src/Platform/iOS/ViewExtensions.cs:281]: Why are we looping through all the Sublayers twice?

**Review by hartez — CHANGES_REQUESTED** [review]
Fix the UpdateBackgroundLayerFrame method.

**Comment by vhugogarcia** [comment]
Thanks @jsuarezruiz for helping to solve this issue. This fix is going to be amazing due to it will help us to reduce the layers in our apps just to render a gradient in the app screens. 👍🏻

File: src/Core/src/Platform/iOS/ViewExtensions.cs (hunk 3 above)
Comment (jsuarezruiz) [reply src/Core/src/Platform/iOS/ViewExtensions.cs:276]: You're right. I have applied changes to remove the unnecessary iteration.

**Review by rmarinho — APPROVED** [review]

**Review by hartez — APPROVED** [review]

**Pull request merged**